### PR TITLE
cn-infra: local file kv db broker (Bolt)

### PR DESCRIPTION
**Pull request by pkotucek**
First draft of kv DB broker (Bolt)
Signed-off-by: Pavel Kotucek <pkotucek@cisco.com>

### CODE REVIEW

File: db/keyval/bolt/bytes_broker_impl.go [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+		boltLogger.SetLevel(logging.DebugLevel)
+	}
+}
+
+// Client serves as a client for Bolt KV storage and implements keyval.CoreBrokerWatcher interface.
+type Client struct {
+	dbPath          *bolt.DB
+	bucketSeparator string
+}
+
```
Comment (lukasmacko) [inline db/keyval/bolt/bytes_broker_impl.go]: Please add a function that creates `Client` based on config

File: db/keyval/bolt/plugin_impl_bolt.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"github.com/ligato/cn-infra/db/keyval/kvproto"
+	"github.com/ligato/cn-infra/flavors/local"
+	"log"
+)
+
+// Config represents configuration for Bolt plugin.
+type Config struct {
+	DbPath          string `json:"db-path"`
+	BucketSeparator string `json:"bucket-separator"`
+}
```
Comment (lukasmacko) [inline db/keyval/bolt/plugin_impl_bolt.go]: add config items to define 
- `FilePermissions` of created db
- `timeout` for opening db

File: db/keyval/bolt/bytes_broker_impl.go [hunk 3]
```diff
@@ -0,0 +1,684 @@
...
+		readonly:  false,
+		separator: client.bucketSeparator,
+		kv:        tx,
+	}
+}
+
+// Create bucket base on given names in current transaction
+func createBucket(tx *bolt.Tx, bucketNames [][]byte) (*bolt.Bucket, error) {
+	var bucket *bolt.Bucket
+	var bucketName []byte
```
Comment (ondrej-fabry) [inline db/keyval/bolt/bytes_broker_impl.go]: You don't need to declare all variables, especially not if used in single scope such as the for loop below. You can use `for _, bucketName := range bucketNames`.

File: db/keyval/bolt/bytes_broker_impl.go [hunk 4]
```diff
@@ -0,0 +1,684 @@
...
+	var err error
+	for _, bucketName = range bucketNames {
+		errBucket = fmt.Sprintf("%s//%s", errBucket, bucketName)
+		if bucket == nil {
+			bucket, err = tx.CreateBucketIfNotExists(bucketName)
+		} else {
+			bucket, err = bucket.CreateBucketIfNotExists(bucketName)
+		}
+		if err != nil {
+			return nil /*err*/, fmt.Errorf("can't create bucket %q", errBucket)
```
Comment (ondrej-fabry) [inline db/keyval/bolt/bytes_broker_impl.go]: The err value should be shown inside the error, other we won't know the reason for the error, only where it happened.

**Comment by ondrej-fabry** [comment]
Closed in favor of #325

**Pull request closed without merging**